### PR TITLE
implemented C# version of mono_class_is_valuetype

### PR DESCRIPTION
**Pull request by jeanclaudegrenier**

### CODE REVIEW

**Review by UnityAlex — APPROVED** [review]
LGTM

File: unity/unity-embed-host/CoreCLRHost.cs [hunk 1]
```diff
@@ -500,6 +500,14 @@ public static IntPtr unity_method_get_function_pointer([NativeCallbackType("Mono
         return IntPtr.Zero;
     }
 
+    [return: NativeCallbackType("gboolean")]
+    public static bool class_is_valuetype(
+        [NativeCallbackType("MonoClass*")] IntPtr klass)
+    {
+        Type t = klass.TypeFromHandleIntPtr();
+        return t?.IsValueType??false;
```
Comment (joncham) [inline unity/unity-embed-host/CoreCLRHost.cs]: Should there be a case where `t` is null?

File: unity/UnityEmbedHost.Tests/BaseEmbeddingApiTests.cs [hunk 2]
```diff
@@ -616,6 +616,16 @@ static void AssertMultiDimensionalArraysAreEquivalent(Array expected, Array actu
         Assert.That(actualFlat, Is.EquivalentTo(expectedFlat));
     }
 
+
+    [TestCase(typeof(Animal),        false)]
+    [TestCase(typeof(ValueMammal),    true)]
+    [TestCase(typeof(Classification), true)]
```
Comment (joncham) [inline unity/UnityEmbedHost.Tests/BaseEmbeddingApiTests.cs]: I would add generic test case. And potentially modified types like a pointer `int*` or by reference `int&`

File: unity/unity-embed-host/CoreCLRHost.cs (hunk 1 above)
Comment (jeanclaudegrenier) [reply unity/unity-embed-host/CoreCLRHost.cs]: good catch, this is a relic from how I converted my MonoClass to Type, before Alex mentioned the TypeFromHandleIntPtr function. will remove the check from related PRs too

**Review by joncham — APPROVED** [review]

**Pull request merged**